backend-lib/src/main/rust/lib.rs

```diff
@@ -146,13 +146,13 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_in
 ///
 /// Returns 0 if successful, or -1 otherwise.
 #[no_mangle]
-pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_initBlockMetaDb(
-    env: JNIEnv<'_>,
-    _: JClass<'_>,
-    fsblockdb_root: JString<'_>,
+pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_initBlockMetaDb<'local>(
+    mut env: JNIEnv<'local>,
```

Comment:
I was confused by this being `mut env` rather than `&mut env`. I guess [this section of the `jni` crate docs](https://docs.rs/jni/latest/jni/struct.JNIEnv.html#cannot-borrow-as-mutable) partially explains it: if this parameter were `&mut env` then it would need to be last. But it is not clear to me *why* that is; or what the trade-offs are between making it `mut env`, and making it `&mut env` with reordered parameters.

Reply:
If it is `mut env` then you always need to create a new `JNIEnv` in the caller like this, correct?
```rust
let env = jvm.attach_current_thread().unwrap();
```